linode/ipv6range/framework_resource.go

```diff
@@ -0,0 +1,224 @@
+package ipv6range
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/linode/helper"
+)
+
+const ipv6rangeNotExist string = "IPv6 range does not exist."
+
+func NewResource() resource.Resource {
+	return &Resource{
+		BaseResource: helper.NewBaseResource(
+			helper.BaseResourceConfig{
+				Name:   "linode_ipv6_range",
+				IDType: types.StringType,
+				Schema: &frameworkResourceSchema,
+			},
+		),
+	}
+}
+
+type Resource struct {
+	helper.BaseResource
+}
+
+func (r *Resource) Create(
+	ctx context.Context,
+	req resource.CreateRequest,
+	resp *resource.CreateResponse,
+) {
+	var data ResourceModel
+	client := r.Meta.Client
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	createOpts := linodego.IPv6RangeCreateOptions{
+		PrefixLength: int(data.PrefixLength.ValueInt64()),
+	}
+
+	if !data.LinodeId.IsNull() && !data.LinodeId.IsUnknown() {
+		createOpts.LinodeID = int(data.LinodeId.ValueInt64())
+	} else if !data.RouteTarget.IsNull() && !data.RouteTarget.IsUnknown() {
+		createOpts.RouteTarget = strings.Split(data.RouteTarget.ValueString(), "/")[0]
+	} else {
+		resp.Diagnostics.AddError(
+			"Failed to create ipv6 range.",
+			"Either linode_id or route_target must be specified.",
+		)
+		return
+	}
+
+	ipv6range, err := client.CreateIPv6Range(ctx, createOpts)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Failed to create ipv6 range.",
+			err.Error(),
```

Comment:
Can we put `linode_id` or `route_target` to be a part of the error message so the user will know which IPv6 range creation was failed?